src/auth/bearer/Token.h

```diff
@@ -0,0 +1,45 @@
+#ifndef _SQUID_SRC_AUTH_BEARER_TOKEN_H
+#define _SQUID_SRC_AUTH_BEARER_TOKEN_H
+
+#if HAVE_AUTH_MODULE_BEARER
+
+#include "auth/bearer/forward.h"
+#include "base/RefCount.h"
+#include "sbuf/SBuf.h"
+#include "SquidTime.h"
+
+namespace Auth {
+namespace Bearer {
+
+/// a Bearer token we have seen and details we associate with it
+class Token : public RefCountable
+{
+    MEMPROXY_CLASS(Auth::Bearer::Token);
+
+public:
+    Token() : expires(squid_curtime) {}
+    explicit Token(const SBuf &token) : b68encoded(token), expires(squid_curtime) {}
+    ~Token();
+
+    /// the B68 encoding form of this token
+    SBuf b68encoded;
+
+    /// the Auth::User this token is tied to
+    Auth::Bearer::User *user = nullptr;
+
+    /// when this token will expire
+    time_t expires = 0;
+
+    /// a cache of known tokens
+    static TokenCache Cache;
+
+private:
+    Token(const Token &); // do not implement
```

Comment:
FWIW, I do not recall any audits prohibiting "deleting" undesirable generated methods in new classes. I also find a bare "I was told to do so" argument rather difficult to accept: Maybe I am missing a very good reason for doing what you are doing; maybe the old audit was flat wrong; maybe the old audit does not apply here; etc. Lots of possibilities here, and you are not giving me enough information to understand which one(s) are relevant, so that I can agree with you and withdraw my comment.

If you believe that some past audit forced you to do something wrong, and you are now using that audit as the reason to continue doing something you think is wrong, then please cite that audit!